### PR TITLE
Shared core platform functionality

### PR DESCRIPTION
**Pull request by YabMas**
Completed core-functionality:
- Items can be synced between boards. Name and status are the only tracked properties.
- There're endpoints for retrieving all linked/unlinked items of a board. 
- There's an endpoint which configures if items should be synced by default for a given board-link

Builds run, but don't work because there's no env-var in the container for the db-config atm. Shouldn't be too hard, but will be done when fixing the config for the whole project. Added a comment to #2.

Most of the core functionality is made platform-agnostic, this aided reuse and shows good code-practices. There's some platform-specific code left, but it's not considered worth it to remove it now because making the core platform-agnostic is not a goal for the project so it's been decided to put effort in the real goals. See #8 

There's a know-bug when updating the name of an item in monday the status gets set to nil because the webhook-req doesn't contain it. The fix shouldn't be too hard, but the current state is adequate for demo-purposes so it's not considered a major concern. See #9

### CODE REVIEW

File: config/.secret.local
Comment (mitchelkuijpers) [inline config/.secret.local:1]: Deze file moet in je gitignore en niet meegecommit worden

File: database.rules.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  /* Visit https://firebase.google.com/docs/database/security to learn more about security rules. */
```
Comment (mitchelkuijpers) [inline database.rules.json:2]: Dit zou ik even weghalen want comments in JSON kan niet

File: prototypes/shared/deps.edn [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+         info.sunng/ring-jetty9-adapter {:mvn/version "0.18.5"}
+         metosin/reitit                 {:mvn/version "0.5.9"}
+         metosin/muuntaja               {:mvn/version "0.6.8"}
+         ring/ring-codec                {:mvn/version "1.2.0"}
+         ; state-mangement
+         mount/mount                    {:mvn/version "0.1.17"}
+         ;domain mapping
+         metosin/malli                  {:mvn/version "0.9.2"}
+         ; testing
+         com.hyperfiddle/rcf            {:mvn/version "20220926-202227"}}}
```
Comment (mitchelkuijpers) [inline prototypes/shared/deps.edn:17]: Graag niet de maps zo alignen dat doen we nergens

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/core/items.clj [hunk 3]
```diff
@@ -0,0 +1,9 @@
+(ns avisi.apps.tech-testing-ground.prototypes.shared.core.items)
+
+(def item-schema
+  [:map
+   {:decode/json #(into {} %)}
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/core/items.clj:5]: Dit zou je kunnen vervangen door de functie `set`

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/jira/domain_mapping.clj [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    [hyperfiddle.rcf :refer [tests]]
+    [malli.core :as m]
+    [malli.transform :as mt]))
+
+(def jira-item-schema
+  [:map
+   [:issue/key string?]
+   [:issue/summary string?]
+   [:issue/description string?]
+   [:issue/status [:enum "To Do" "In Progres" "Done"]]])
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/jira/domain_mapping.clj:14]: Ik zou hier iets doen zoals: `(into [:enum] (keys domain-item-keys->jira-item-keys))` Dan ben je die duplicatie kwijt

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/jira/routes.clj [hunk 5]
```diff
@@ -32,22 +42,36 @@
...
-             :headers {"content-type" "application/json"}
-             :body descriptor})}}]
-   ["/atlassian/lifecycle/:lifecycle"
+  [["/atlassian/lifecycle/:lifecycle"
     {:post
        {:parameters {:body map?}
         :handler
-          (fn [_]
+          (fn [req]
+            (def lc_req req)
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/jira/routes.clj:50]: Dit mag weg denk ik

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj [hunk 6]
```diff
@@ -0,0 +1,133 @@
+(ns avisi.apps.tech-testing-ground.prototypes.shared.platforms.monday.api-wrapper
+  (:require
+    [avisi.apps.tech-testing-ground.prototypes.shared.peripherals.http-client :as http-client]
+    [clojure.data.json :as json]))
+
+(def ^:private api-url "https://api.monday.com/v2/")
+
+(def ^:private sent-query
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj:8]: `send-query!` zou ik een logischeren naam vinden `sent` is verleden tijd.

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+  ; to stick with pure gql-queries. These queries are documented here for now in case reasons arise to move to the lib
+  ; after all.
+  [{:boards [:id :name]}]
+  [{(:boards {:ids [~board-id]}) [{:items [:id :name]}]}]
+  [{('create_item
+     {:board_id ~board-id
+      :item_name ~item-name
+      :column_values ~column-values})
+      [:id]}]
+  [{('delete_item {:item_id ~item-id}) [:id]}])
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj:133]: Ik zou deze keuze heroverwegen omdat je nu strings aan elkaar aan het plakken bent kan je issue krijgen met dingen zoals `"{..}"` als input

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/webhooks.clj [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+   :board-id board-id
+   :item
+     (->
+       (webhook-event->monday-item event)
+       (domain/monday-item->domain-item))
+   :action (webhook-event-type->domain-action event-type)})
+
+(def propagate-action (propagate/propagate-action-fn webhook-req->propagation-args))
+
+(defn webhook-handler [req] (def _r req) (try (propagate-action req) {:status 200} (catch Exception _ {:status 500})))
```
Comment (mitchelkuijpers) [inline prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/webhooks.clj:52]: die def _r kan weg neem ik aan

**Review by mitchelkuijpers — APPROVED** [review]

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj (hunk 7 above)
Comment (YabMas) [reply prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj:133]: I dont need to concatenate strings as I use graph-qls native way of substituting variables. I think it's fine for this usecase and going with the clojure-datastructure option only starts to make sense when you need to dynamically compose the query itself. I might need a rational for this decision later though.

File: prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/core/items.clj (hunk 3 above)
Comment (YabMas) [reply prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/core/items.clj:5]: `set` results in a hash-set, I need to turn a java.util.Hashmap into a clojure-map. I believe this is the easiest way.

Comment (YabMas) [reply prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/core/items.clj:5]: On a second look I can actually remove this line here as the problem is now being solved elsewhere.

**Pull request merged**